tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetSetService.scala

```diff
@@ -0,0 +1,40 @@
+package com.linagora.tmail.james.jmap.publicAsset
+
+import jakarta.inject.{Inject, Named}
+import org.apache.commons.configuration2.Configuration
+import org.apache.james.jmap.api.model.{IdentityId, UploadId}
+import org.apache.james.jmap.core.AccountId
+import org.apache.james.jmap.mail.BlobId
+import org.apache.james.jmap.method.IdentityResolver
+import org.apache.james.mailbox.MailboxSession
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import scala.util.Try
+
+class PublicAssetSetService @Inject()(val identityResolver: IdentityResolver,
+                                      @Named("jmap") jmapConfiguration: Configuration) {
+
+  private val publicURIPrefix: String = jmapConfiguration.getString("publicAsset.publicURIPrefix", "")
+  private val uploadIdPrefix = "uploads-"
+
+  def generatePublicURI(publicAssetId: PublicAssetId, accountId: AccountId): PublicURI =
+    PublicURI.from(publicAssetId, accountId, publicURIPrefix)
+
+  def checkIdentityIdsExist(identityIds: Seq[IdentityId], session: MailboxSession): SMono[Seq[IdentityId]] =
+    SFlux.fromIterable(identityIds)
+      .concatMap(identityId => checkIdentityIdExist(identityId, session))
+      .collectSeq()
+
+  private def checkIdentityIdExist(identityId: IdentityId, session: MailboxSession): SMono[IdentityId] =
+    identityResolver.resolveIdentityId(identityId, session)
+      .flatMap(SMono.justOrEmpty)
+      .map(_ => identityId)
+      .switchIfEmpty(SMono.error(PublicAssetIdentityIdNotFoundException(identityId.serialize)))
+
+  def blobIdFromUploadId(uploadId: UploadId): Try[BlobId] = BlobId.of(uploadIdPrefix + uploadId.asString())
+
+  def uploadIdFromBlobId(blobId: BlobId): Try[UploadId] = blobId.value.value match {
+    case value if value.startsWith(uploadIdPrefix) => Try(UploadId.from(value.substring(uploadIdPrefix.length)))
+    case value => throw PublicAssetInvalidBlobIdException(value)
+  }
```

Comment:
1. This knowlege is diplicated with Download routes.

I would expect that we have a single definition and avoid duplication / coupling.

This code smell likely comes from the fact that the code explicitly mandate we work from an `Upload`.

This is likely unecessary.

This coupling is completly avoided if  we rather call BlobResolvers and get a Blob payload. This would also lead to more flexible code and better stick to the spec. After all the spec says so far "blobId" and do not mention this should be an upload. I would then expect it works with anything that is a blobId!